feat: rework Task Manager Real Data

Stijnus opened the pull request:
Bugfix for : https://github.com/stackblitz-labs/bolt.diy/issues/1455

# TaskManager Tab Changes

## Core Functionality Updates
- Removed JavaScript heap memory monitoring in favor of system memory metrics
- Added real system memory monitoring via `/api/system/memory-info` endpoint
- Implemented swap memory monitoring and visualization
- Enhanced network latency monitoring with real measurements
- Added historical data tracking for metrics (last 30 measurements)

## Performance Improvements
- Reduced update interval from 1s to 2.5s to decrease overhead
- Added tab visibility detection to pause updates when tab is not visible
- Implemented proper cleanup of Chart.js instances
- Limited metrics history to prevent memory buildup
- Optimized chart rendering with React.useMemo

## UI/UX Enhancements
- Added separate cards for System Memory and Swap Memory
- Improved memory usage visualization with color-coded indicators
- Enhanced network metrics display (current, min, max, average latency)
- Added tooltips for technical terms
- Implemented responsive progress bars for memory usage

## Alert System
- Implemented smart alert system that only triggers on state changes
- Added critical memory usage notifications (>95% threshold)
- Limited alerts history to last 10 entries
- Added toast notifications with 5s auto-close
- Implemented "Clear All" functionality for alerts

## Type Safety
- Added proper TypeScript interfaces for all metrics
- Implemented proper type checking for system memory info
- Added proper error handling for API calls
- Enhanced type safety for Chart.js references

## Code Organization
- Separated metric collection logic into discrete functions
- Implemented proper cleanup patterns
- Added comprehensive error handling
- Improved code readability with proper spacing and organization

## Screenshots

![screencapture-localhost-5173-2025-03-08-23_12_47](https://github.com/user-attachments/assets/4e46b063-34d8-4923-9218-3ecdd9c72998)

leex279 commented:
I am not sure if is correct. For me, at least with pnpm run dev, it spawns 2 nodejs nodes, and you show only the usage of the lower one
![image](https://github.com/user-attachments/assets/07cae5c6-5c25-4821-bfc3-1aae33972be3)

EXELANCE-LLC commented:
40 kb seems to be total Memory.  Where can we raise this?

thecodacus commented:
I would like to make this feature enabled/disabled based on environment variables. 
for infra like netlify cloudflare these metrics will not be available. so this should be by default disabled and only if the user wants and running in a VM or server then should enable from environment vars

Stijnus commented:
> I would like to make this feature enabled/disabled based on environment variables. for infra like netlify cloudflare these metrics will not be available. so this should be by default disabled and only if the user wants and running in a VM or server then should enable from environment vars

I will check it out if we can implement it...

Stijnus commented:
I have made the changes : 

TaskManager Environment Detection Enhancement
Added intelligent detection for serverless environments (Cloudflare, Netlify, Vercel) where system monitoring APIs are unavailable.

Changes
Added detection for common serverless hosting platforms where system monitoring isn't supported
Added fallback UI with an informative message explaining why monitoring is disabled
Implemented API availability checks to avoid unnecessary requests in unsupported environments
Added testing tools for developers to simulate different environments

Why This Matters
The TaskManager component requires access to system-level metrics (CPU, memory, disk) which aren't available in serverless environments. This change improves the user experience by:
Preventing failed API requests in environments where they'll never succeed
Providing clear feedback about why the feature isn't available
Ensuring the TaskManager works optimally on supported platforms (local development, VMs, dedicated servers)

Testing
When running in development mode, you can test different scenarios by:
Normal view: Access without parameters
Simulate serverless environment: Add ?simulate-serverless=true to URL
Simulate API failures: Add ?simulate-api-failure=true to URL

<img width="1800" alt="image" src="https://github.com/user-attachments/assets/8bbfc660-aea6-43c7-a0c7-e063ca82e1d2" />
